include/libtorrent/span.hpp

```diff
@@ -60,10 +60,12 @@ namespace libtorrent
 		span(U (&arr)[N])
 			: m_ptr(&arr[0]), m_len(N) {}
 
+#if (defined __GNUC__ || (defined _MSC_VER && _MSC_VER >= 1900))
 		// anything with a .data() member function is considered a container
 		template <typename Cont, typename = decltype(std::declval<Cont>().data())>
```

Comment:
I tried with no success, I even tried to recreate it with another construction, the problem is that swig fail to parse if `decltype` has any non trivial expression inside